File: pet-store-repository/src/main/java/com/chewy/osn/repository/SpeciesAttributeConverter.java
package com.chewy.osn.repository;

import com.chewy.osn.domain.Species;
import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
import software.amazon.awssdk.enhanced.dynamodb.AttributeValueType;
import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
import software.amazon.awssdk.services.dynamodb.model.AttributeValue;

public class SpeciesAttributeConverter implements AttributeConverter<Species> {

    @Override
    public AttributeValue transformFrom(Species input) {
        return AttributeValue.builder()
            .s(input.toString())
            .build();
    }

    @Override
    public Species transformTo(AttributeValue input) {
        return Species.valueOf(input.s());
    }

    @Override
    public EnhancedType<Species> type() {
        return EnhancedType.of(Species.class);
    }

    @Override
    public AttributeValueType attributeValueType() {
        return AttributeValueType.S;
    }
}

File: pet-store-dynamo-repository/src/main/kotlin/com/johnkruss/osn/dynamo/SpeciesAttributeConverter.kt
package com.johnkruss.osn.dynamo

import com.johnkruss.osn.domain.Species
import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter
import software.amazon.awssdk.enhanced.dynamodb.AttributeValueType
import software.amazon.awssdk.enhanced.dynamodb.EnhancedType
import software.amazon.awssdk.services.dynamodb.model.AttributeValue

class SpeciesAttributeConverter : AttributeConverter<Species> {
    override fun transformFrom(input: Species): AttributeValue {
        return AttributeValue.builder()
            .s(input.toString())
            .build()
    }

    override fun transformTo(input: AttributeValue): Species {
        return Species.valueOf(input.s())
    }

    override fun type(): EnhancedType<Species> {
        return EnhancedType.of(Species::class.java)
    }

    override fun attributeValueType(): AttributeValueType {
        return AttributeValueType.S
    }
}
